pytorch_lightning/plugins/precision/cpu_native_amp.py

```diff
@@ -0,0 +1,46 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from pytorch_lightning.plugins.precision.native_amp import NativeMixedPrecisionPlugin
+from pytorch_lightning.utilities import _TORCH_CPU_AMP_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class CPUNativeMixedPrecisionPlugin(NativeMixedPrecisionPlugin):
+    """
+    Plugin for CPU native mixed precision training with :mod:`torch.cpu.amp`.
+
+    Args:
+        precision: Currently only bf16 (torch.bfloat16) is supported.
+    """
+
+    def __init__(self, precision: str = "bf16") -> None:
+        if not _TORCH_CPU_AMP_AVAILABLE:
+            raise MisconfigurationException(
+                "You have asked for native AMP on CPU, but AMP is only available on GPU for PyTorch 1.9 "
+                "and lower. To use native AMP on CPU, install PyTorch 1.10 or later."
+            )
+        super().__init__(precision)
+
+    def _select_precision_dtype(self, precision: str = "bf16") -> torch.dtype:
+        if not precision == "bf16":
+            raise MisconfigurationException(
+                "CPU native amp only supports bfloat16. Please pass precision=bf16 to the Trainer."
+            )
+        return torch.bfloat16
+
+    def autocast_context_manager(self) -> torch.cuda.amp.autocast:
+        return torch.cpu.amp.autocast(fast_dtype=self._fast_dtype)
```

Comment:
we can't add this to NativeMixedPrecisionPlugin?

Reply:
We could totally by introducing a 'device' flag to the NativeMixedPrecisionPlugin, however that would involve some API work + enforcement of logic when on CPU (only bf16 supported). I felt like separating out the responsibilities since it's a completely different call would be reasonable!

the RFC is moving towards having it being all in one context manager: https://github.com/pytorch/pytorch/issues/55374 it seems like for now the API is split between cuda/cpu.

If anyone has any hard feelings here happy to swap between as it makes little difference to the user, primarily for lightning developers!

Reply:
I just think that inheritance is not right here, because NativeMixedPrecisionPlugin already is an implementation. When you did the bfloat16 you could have also made a BfloatNativeMixedPrecision the same way, it just whould have required overriding more methods. 

since native means pytorch, and pytorch calls either into torch.cuda or torch.cpu, it is more reasonable to me to bring it under the same plugin. the only information we would need is whether we are on cpu or on cuda, but this we know at the time of instantiation in the accelerator connector, is my thinking correct?

Reply:
I'm actually surprised @ananthsub did not request changes here as he is normally more concerned with this than me 😅

Reply:
Sure! I saw this coming initially but was curious if anyone would care/realise on this path. My initial thinking was cpu autocast involves completely different function calls, whereas bfloat was a dtype that can be passed to the plugin (hence adding the the same behaviour to the native precision plugin)!

More than happy to make the change to unblock this feature :)

Reply:
what does @justusschock think?

Reply:
@awaelchli @SeanNaren tbh my hope is once we better spec out #7324 we can assess this streamlining then (@four4fish ).